### PR TITLE
Support passing and handling SIP headers.

### PR DESCRIPTION
**Pull request by dennwc**
Support passing and handling SIP headers. Also makes tests more robust and fixes integration tests on Mac.

Requires https://github.com/livekit/protocol/pull/805, https://github.com/livekit/livekit/pull/2993

### CODE REVIEW

File: pkg/sip/client.go [hunk 1]
```diff
@@ -128,11 +128,19 @@ func (c *Client) CreateSIPParticipant(ctx context.Context, req *rpc.InternalCrea
 	} else if req.RoomName == "" {
 		return nil, fmt.Errorf("room name must be set")
 	}
-	log := c.log.WithValues(
+	log := c.log
+	if req.ProjectId != "" {
+		log = log.WithValues("project", req.ProjectId)
```
Comment (davidzhao) [inline pkg/sip/client.go]:
```suggestion
		log = log.WithValues("projectID", req.ProjectId)
```

File: pkg/sip/inbound.go [hunk 2]
```diff
@@ -146,13 +146,19 @@ func (s *Server) onInvite(req *sip.Request, tx sip.ServerTransaction) {
...
-	r, err := s.handler.GetAuthCredentials(ctx, from.User, to.User, to.Host, src)
+	r, err := s.handler.GetAuthCredentials(ctx, callID, from.User, to.User, to.Host, src)
 	if err != nil {
 		cmon.InviteErrorShort("auth-error")
 		log.Warnw("Rejecting inbound, auth check failed", err)
 		cc.RespondAndDrop(sip.StatusServiceUnavailable, "Try again later")
 		return
 	}
+	if r.ProjectID != "" {
+		log = log.WithValues("project", r.ProjectID)
```
Comment (davidzhao) [inline pkg/sip/inbound.go]: same as above
```suggestion
		log = log.WithValues("projectID", r.ProjectID)
```

File: pkg/sip/inbound.go [hunk 3]
```diff
@@ -269,6 +276,9 @@ func (c *inboundCall) handleInvite(ctx context.Context, req *sip.Request, tx sip
 		Pin:        "",
 		NoPin:      false,
 	})
+	if disp.ProjectID != "" {
+		c.log = c.log.WithValues("project", disp.ProjectID)
```
Comment (davidzhao) [inline pkg/sip/inbound.go]:
```suggestion
		c.log = c.log.WithValues("projectID", disp.ProjectID)
```

File: pkg/sip/inbound.go [hunk 4]
```diff
@@ -471,6 +494,9 @@ func (c *inboundCall) pinPrompt(ctx context.Context) (disp CallDispatch, _ bool)
 					Pin:        pin,
 					NoPin:      noPin,
 				})
+				if disp.ProjectID != "" {
+					c.log = c.log.WithValues("project", disp.ProjectID)
```
Comment (davidzhao) [inline pkg/sip/inbound.go]:
```suggestion
					c.log = c.log.WithValues("projectID", disp.ProjectID)
```

**Review by davidzhao — COMMENTED** [review]
just a nit about logging field

File: pkg/sip/client.go (hunk 1 above)
Comment (dennwc) [reply pkg/sip/client.go]: I think I've seen it spelled `project` in another repository. Will find that place and will update it too then 👍

Comment (dennwc) [reply pkg/sip/client.go]: Done

**Review by paulwe — APPROVED** [review]

**Pull request merged**